pkg/registry/apis/provisioning/jobs/worker.go

```diff
@@ -170,7 +174,36 @@ func (g *JobWorker) Process(ctx context.Context, job provisioning.Job) (*provisi
 		if err := commenter.Process(ctx, job); err != nil {
 			return nil, fmt.Errorf("error processing pull request: %w", err)
 		}
+
 	case provisioning.JobActionExport:
+		if job.Spec.Export != nil && job.Spec.Export.History {
```

Comment:
A couple of points on this one: 
- JobWorker should remain as lean as possible because it already has a lot of the wiring and selection on which type of job and what to call. 
- The Export logic should be condensed in a single spot where the actual logic is implemented. This comment is a code smell`// Uses k8s API against the configured folder`. I would plug the `LegacyExporter` into the replicator as well and the export options in the signature of that `Export` method and handle the cases there.